### PR TITLE
[CELEBORN-682] Master and client should  handle blacklist worker and shutting down worker separately

### PR DESCRIPTION
**Pull request by AngersZhuuuu**

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @RexXiong @waitinfuture @FMX

File: client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala [hunk 1]
```diff
@@ -253,6 +253,7 @@ class ChangePartitionManager(
         .keySet()
         .asScala
         .filter(w => !lifecycleManager.workerStatusTracker.blacklist.keySet().contains(w))
```
Comment (RexXiong) [inline client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala:255]: It's nice if encapsulate these filter logic into workerStatusTracker

File: client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala [hunk 2]
```diff
@@ -693,6 +693,9 @@ class LifecycleManager(appId: String, val conf: CelebornConf) extends RpcEndpoin
           failureInfos.add(s"[reserveSlots] Failed to" +
             s" reserve buffers for ${Utils.makeShuffleKey(applicationId, shuffleId)}" +
             s" from worker ${workerInfo.readableAddress()}. Reason: ${res.reason}")
+          if (res.status.equals(StatusCode.WORKER_SHUTDOWN)) {
+            workerStatusTracker.resolveShutdownWorkers(List(workerInfo).asJava)
```
Comment (RexXiong) [inline client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala:697]: Do we need notify the listeners this time？

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala [hunk 3]
```diff
@@ -736,7 +737,7 @@ private[celeborn] class Master(
     context.reply(CheckQuotaResponse(isAvailable, reason))
   }
 
-  private def workersNotBlacklisted(
+  private def workersAvailable(
```
Comment (RexXiong) [inline master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala:740]: Probably workersAvailable need consider shutdown workers every time? so filter with shutdownWorkerSnapshot together with blacklist, then param  may for other Scenario.

Comment (AngersZhuuuu) [reply master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala:740]: Yea, done

File: client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala (hunk 2 above)
Comment (AngersZhuuuu) [reply client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala:697]: Not so necessary, remove it.

File: client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala (hunk 1 above)
Comment (AngersZhuuuu) [reply client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala:255]: Done

**Comment by AngersZhuuuu** [comment]
ping @RexXiong @waitinfuture  Any more suggestion?

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main/0.3

**Pull request closed without merging**